docs/source/model/python_model.rst

```diff
@@ -0,0 +1,437 @@
+``python_function`` flavor that contain user-specified code and *artifact* (file) dependencies.
+The MLflow PythonModel enables you to implement custom model logic while leveraging MLflow’s 
+packaging and deployment capabilities.There are two ways to define a PythonModel: 
+Subclassing :py:class:`mlflow.pyfunc.PythonModel <mlflow.pyfunc.PythonModel>` or defining a callable.
+This guide provides a complete walkthrough on how to define and use a custom PythonModel.
+
+Define a custom PythonModel
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Option 1: Subclass PythonModel
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+:py:mod:`mlflow.pyfunc` module provides a :py:class:`generic PythonModel class <mlflow.pyfunc.PythonModel>` that can be used to define your own 
+customized model. By subclassing it, the model can be seamlessly integrated with other MLflow components.
+
+Methods of PythonModel:
+
+    - **predict**
+        A valid PythonModel must implement the predict method, which defines the model’s prediction logic. This method is called by MLflow when 
+        the model is loaded as a PyFunc model using ``mlflow.pyfunc.load_model`` and the ``predict`` function is invoked.
+    - **predict_stream**
+        The predict_stream method should be implemented if the model is intended for use in streaming environments. MLflow invokes this method 
+        when the model is loaded as a PyFunc model with ``mlflow.pyfunc.load_model`` and ``predict_stream`` is called.
+    - **load_context**
+        Implement the load_context method if the model requires additional context to be loaded. For more details, refer to :py:func:`load_context() <mlflow.pyfunc.PythonModel.load_context>`.
+
+.. tip::
+    Starting from MLflow 2.20.0, ``context`` parameter can be removed from ``predict`` and ``predict_stream`` functions if it is not used.
+    e.g. ``def predict(self, model_input, params)`` is a valid predict function signature.
+
+Below is an example of a simple PythonModel that takes a list of string and returns it.
+
+.. code-block:: python
+
+    import mlflow
+
+
+    class MyModel(mlflow.pyfunc.PythonModel):
+        def predict(self, model_input: list[str], params=None) -> list[str]:
+            return model_input
+
+
+
+Option 2: Define a callable
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+An alternative way to log a PythonModel is to define a callable that **takes a single argument** and returns a prediction. This callable can be
+logged as a PythonModel by passing it to ``mlflow.pyfunc.log_model``.
+
+.. tip::
+    Starting from MLflow 2.20.0, you can use @pyfunc decorator on the callable to enable data validation on the input based on the type hints.
+    Check :ref:`type hint usage in PythonModel <type-hint-usage-in-pythonmodel>` for more details.
+
+.. code-block:: python
+
+    def predict(model_input: list[str]) -> list[str]:
+        return model_input
+
+Log the model
+^^^^^^^^^^^^^
+PythonModel should be logged as python_function flavor using :py:func:`mlflow.pyfunc.log_model`.
```

Comment:
I wouldn't say "python_function flavor" - perhaps just say that we're using the pyfunc module to directly log a custom model.